honey-badger.py

```diff
@@ -14,25 +14,53 @@
     path to the scripts are hard-coded into the *.ghpy file.
 
     The -i switch installs the honey-badger kit after building to the user's Grasshopper Library folder.
+    
+    The -v parameter defines the Rhino version to find / add RhinoCommon and Grasshopper DLLs from.
+    
+    The -s switch skips a compatibility check to see that IronPython version matches the Rhino version desired (note: Rhino plugins are, for now, forward but not backward compatible!)
 """
 
 import sys
 import os
+import platform
 import clr
 import json
 import argparse
 import string
 import shutil
 import traceback
-import parameter_compiler
 
 clr.AddReference("System")
 clr.AddReference("System.IO")
 import System
 import System.IO
 
 
-def main(badger_file, editable, install):
+def main(badger_file, editable, install, rhino_version, skip_compatibility_check):
+    
+    if not skip_compatibility_check:
+        rhino_to_ipy_version = {
+            '6': ['2.7.8'],
+            '7': ['2.7.8', '2.7.9']
+        }
+        
+        ipy_version_actual = '{}.{}.{}'.format(sys.version_info.major, sys.version_info.minor, sys.version_info.micro)
+        ipy_version_allowed = rhino_to_ipy_version[rhino_version]
+        # Checks that IronPython is being used
+        assert platform.python_implementation() == 'IronPython', "You need to use IronPython to badger your components. Currently using {}".format(platform.python_implementation())
+        assert ipy_version_actual in ipy_version_allowed, \
+            "IronPython version does not match for Rhino {rhino_version}. Must be {version_target} but was {version_actual}".format(
+                rhino_version=rhino_version,
+                version_target=' or '.join(ipy_version_allowed),
+                version_actual=ipy_version_actual
+                )
+        
+        assert os.path.exists("C:\\Program Files\\Rhino {v}\\Plug-ins\\Grasshopper\\Grasshopper.dll".format(v=rhino_version)), "Could not find Grasshopper.dll. Is Rhino {v} installed?".format(v=rhino_version)
```

Comment:
I suggest using the environment variable `%ProgramFiles%` (`os.path.expandvars`? something like that) instead, as some machines will have this located somewhere else (e.g., the `D:` drive)

Reply:
right will do!